test/MathOptInterface/MOI_Wrapper.jl

```diff
@@ -583,4 +583,125 @@ end
+    #   (This is mostly done to be sure no status since the optimal solve
+    #   was kept.)
+    # * MIP-start the model with the optimal solution.
+    # * If the solution found is the MIP-started solution then XPRESS has
+    #   used the MIP-start (we checked XPRESS could not solve it using
+    #   a single node).
+    atol = rtol = 1e-6
+    weight = Float64[
+        677, 460, 752, 852, 580, 116, 457, 121, 454, 870,
+        443, 196, 411, 539, 348, 187, 771, 127, 338, 527,
+        932, 961,  48, 135, 367, 998, 363, 153, 921, 578,
+        311, 560, 293, 258, 474, 884, 162, 136, 479, 289,
+        813, 139, 795, 825, 945, 750, 462, 659, 270,  34,
+        758, 865, 238, 367, 444, 116,  69, 894, 584,  96,
+         29, 199, 712, 703, 856, 692, 396, 409, 603, 632,
+        479, 848, 822, 248, 424, 978, 738, 655, 210, 173,
+        731, 100, 889, 195, 245, 329, 446,  47, 235,  25,
+        254, 150, 520, 665, 391, 907, 123, 826, 959, 176
+    ]
+    profit = weight .- (5.0,)
+    capacity = 10000.0
+    model = Xpress.Optimizer(
+        OUTPUTLOG    = 0,
+        HEURSTRATEGY = 0,
+        CUTSTRATEGY  = 0,
+        PRESOLVE     = 0,
+        MIPPRESOLVE  = 0
+    )
+    # The variables: x[1:100], Bin
+    x, _ = MOI.add_constrained_variables(model, fill(MOI.ZeroOne(), 100))
+    # The objective function: maximize sum(profit' * x)
+    objf = MOI.ScalarAffineFunction(
+        MOI.ScalarAffineTerm.(profit, x), 0.0
+    )
+    MOI.set(
+        model,
+        MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+        objf,
+    )
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    # The capacity constraint: sum(weight' * x) <= capacity
+    MOI.add_constraint(
+        model, MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(
+            weight, x), 0.0
+        ),
+        MOI.LessThan(capacity)
+    )
+
+    # FIRST RUN: get the optimal value. Check that the optimal
+    # value was not discovered at the root node.
+    MOI.optimize!(model)
+
+    solution = MOI.get(model, MOI.VariablePrimal(), x)
+    computed_obj_value = profit' * solution
+    obtained_obj_value = MOI.get(model, MOI.ObjectiveValue()) :: Float64
+    @test isapprox(
+        obtained_obj_value, computed_obj_value; rtol = rtol, atol = atol
+    )
+    @test isapprox(9945.0, computed_obj_value; rtol = rtol, atol = atol)
+    # Using MOI 9.22, so MOI.RawOptimizerAttribute was not available.
+    node_solution_was_found = Xpress.getintattrib(
+        model.inner, Xpress.Lib.XPRS_MIPSOLNODE
+    ) :: Cint
+    #=
+    MOI.get(
+        model, MOI.RawOptimizerAttribute("MIPSOLNODE")
+    ) :: CInt
+    =#
+    @test node_solution_was_found > 1
+
+    # SECOND RUN: run without MIP-start and only searching the first node.
+    # Should give a worse solution than the previous one.
+    MOI.set(model, MOI.RawParameter("MAXNODE"), 1)
+
+    MOI.optimize!(model)
+    @show model.callback_state

```

Comment:
is this needed?

Reply:
Sorry, I left it by mistake. I was just checking why the error 707 was happening. In the rest of the code this field is used to check if the optimization is in progress, so I checked it to see if it had an wrong value. Will remove.